### PR TITLE
Treat thread oops with better care to avoid silent panics

### LINKED ISSUE #1477: Asterinas exits quietly when panic
### Your Question

Asterinas exits quietly when a panic is triggered, is it a new feature? 🤔

### What You Have Tried

For example, if we run `ping www.baidu.com`, because of the `todo!` in 
https://github.com/asterinas/asterinas/blob/998869d57e0e35929348d41f7be19aab866c1f02/kernel/src/util/net/options/mod.rs#L130-L137
A panic is triggered and Asterinas exited with some error info such as stack info. 
However, now Asterinas exits quietly (master 998869d)
image

### PR DESCRIPTION
**Pull request by junyang-zh**
fix #1477

### CODE REVIEW

**Comment by junyang-zh** [comment]
@boterinas ready

File: kernel/src/thread/mod.rs [hunk 1]
```diff
@@ -46,9 +49,11 @@ impl Thread {
         priority: Priority,
         cpu_affinity: CpuSet,
     ) -> Self {
+        let catch_panics = !oops::panic_on_oops();
```
Comment (tatetian) [inline kernel/src/thread/mod.rs:52]: This is a bit of surprise to me. Why is it more preferable to decide the panic-on-oops behavior upon the creation of a thread?

**Review by tatetian — COMMENTED** [review]
The code looks good. But why is this PR assigned the P1 priority?

**Comment by junyang-zh** [comment]
> The code looks good. But why is this PR assigned the P1 priority?

because unfortunately all panics after #1379 are silent, which makes developers astonished.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**